### DISCUSSION TITLE
SKD ERROR

### DISCUSSION THREAD

**Discussion by salisou** [CommunityToolkit/Maui#1336] [Q&A] [answered]
Gravità	Codice	Descrizione	Progetto	File	Riga	Stato eliminazione
Errore	NETSDK1141	non è possibile risolvere la versione di .NET SDK come specificato nel file global.json presente in D:\.Net Maui 2023\Cours Udemy\CommuintyToolkit\global.json.	CommunityToolkit.Maui.Sample	C:\Program Files\dotnet\sdk\7.0.400-preview.23274.1\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets	127

**Comment by VladislavAntonyuk** [comment]
@brminnick that is the reason, why I suggest to remove global.json
@salisou deleting global.json should solve the issue

**Reply by TheCodeTraveler** [reply]
NETSDK1141 means that the user doesn't have the correct version of .NET installed:

https://learn.microsoft.com/dotnet/core/tools/sdk-errors/netsdk1141

Deleting `global.json` will remove the error, but the best approach is to update the .NET SDK installed on your machine.

**Reply by VladislavAntonyuk** [reply]
I have 7.0.400 and .net 8 and while global.json is set 7.0.300 for example, I cannot build the project. Maybe that is my local issue, but that is what I have.

**Reply by VladislavAntonyuk** [reply]
Also with .net 8 as .net MAUI now uses NuGet package again, do we still need global.json?

**Reply by TheCodeTraveler** [reply]
> Also with .net 8 as .net MAUI now uses NuGet package again, do we still need global.json?

Yes. We will update our minimum dependency to .NET 8 for the Toolkits, and we will also then update the minimum .NET version in `global.json` accordingly to ensure developers using our repo are using .NET 8. 

As far as I'm aware, there's no way to ensure that a developer who downloads your code is using a minimum version of .NET without `global.json`.

**Reply by TheCodeTraveler** [reply, marked as answer]
> I have 7.0.400 and .net 8 and while global.json is set 7.0.300 for example, I cannot build the project. 

I haven't tested this to confirm, but here's my hunch: our `global.json` includes `  "rollForward": "latestMajor"` + `"allowPrerelease": false` and you have .NET 8 prerelease installed on your machine. 

.NET _should_ use the `latestMajor` non-prerelease version of .NET installed on your machine, but it sounds like it sees that the newest version of .NET is a prerelease and fails instead. 

This sounds like a .NET bug. 

I'll look into it to see if there's a solution.

**Reply by pictos** [reply]
> Also with .net 8 as .net MAUI now uses NuGet package again, do we still need global.json?

just the preview bits, the stable channel will still use workloads

**Reply by salisou** [reply]
> @brminnick that is the reason, why I suggest to remove global.json
> @salisou deleting global.json should solve the issue

I configured it like this and it works 

{
  "sdk": {
    "version": "7.0.200",
    "rollForward": "latestMajor"
  }
}

**Reply by salisou** [reply]
Thank for the assistance 😀😀🙂

**Reply by TheCodeTraveler** [reply]
@salisou could you also try this configuration? 👇

If this updated config works on your machine, then I'll open a PR to update our `global.json`:

```json
{
  "sdk": {
    "version": "7.0.200",
    "rollForward": "latestFeature",
    "allowPrerelease": false
  }
}
```

**Reply by salisou** [reply]
> @salisou could you also try this configuration? 👇
> 
> If this updated config works on your machine, then I'll open a PR to update our `global.json`:
> 
> ```json
> {
>   "sdk": {
>     "version": "7.0.200",
>     "rollForward": "latestFeature",
>     "allowPrerelease": false
>   }
> }
> 


I configured it like this and it works  

{
>   "sdk": {
>     "version": "7.0.200",
>     "rollForward": "latestFeature",
>     //"allowPrerelease": false
>   }
> }

**Reply by TheCodeTraveler** [reply]
<img width="825" alt="image" src="https://github.com/CommunityToolkit/Maui/assets/13558917/cc772264-a252-40ad-bfcf-82d6513cae82">


Thanks @salisou - I'm just a bit confused by your most recent reply (screen shot above) because it includes both your new configuration and my new configuration. 

Can you confirm that keeping `"allowPrerelease": false` and changing `"rollForward": "latestMajor"` -> `"rollForward": "latestFeature"` in `global.json` resolves the error on your local machine?

**Reply by salisou** [reply]
Yess

**Comment by CliffAgius** [comment]
Closed as answered
